rosidl_generator_c/src/string_functions.c

```diff
@@ -84,14 +82,15 @@ rosidl_generator_c__String__assignn(
   if (n == SIZE_MAX) {
     return false;
   }
-  char * data = (char *)malloc(n + 1);
-  if (!data) {
+  str->data = realloc(str->data, n + 1);
```

Comment:
One thing to keep in mind with realloc(); if it fails, it does *not* free the original memory.  If this fails, I think what will happen here is that str->data will be NULL, and thus whatever old memory was in str->data will be leaked.  You may want to assign this to a temporary to avoid that situation.